components/TextControls/examples/supplemental/MDCTextControlConfiguratorExample.m

```diff
@@ -0,0 +1,204 @@
+    NSArray *traitCollections = @[ currentTraitCollection, contentSizeCategoryTraitCollection ];
+    UITraitCollection *traitCollection =
+        [UITraitCollection traitCollectionWithTraitsFromCollections:traitCollections];
+    [self setOverrideTraitCollection:traitCollection forChildViewController:viewController];
+    [self.view setNeedsLayout];
+  }
+}
+
+- (UIContentSizeCategory)contentSizeCategoryForViewController:(UIViewController *)viewController {
+  if (@available(iOS 10.0, *)) {
+    return viewController.traitCollection.preferredContentSizeCategory;
+  }
+  return nil;
+}
+
+- (void)setContainerScheme:(id<MDCContainerScheming>)containerScheme {
+  self.contentViewController.containerScheme = containerScheme;
+}
+
+- (id<MDCContainerScheming>)containerScheme {
+  return self.contentViewController.containerScheme;
+}
+
+- (CGFloat)preferredContentMinY {
+  if (@available(iOS 11.0, *)) {
+    return (CGFloat)(self.view.safeAreaInsets.top);
+  } else {
+    return (CGFloat)self.topLayoutGuide.length;
+  }
+}
+
+#pragma mark - MDCTraitEnvironmentChangeDelegate
+
+- (void)
+    childViewControllerDidRequestPreferredContentSizeCategoryDecrement:
+        (UIViewController *)childViewController
+                                                        decreaseButton:(MDCButton *)decreaseButton
+                                                        increaseButton:(MDCButton *)increaseButton {
+  [self decreaseContentSizeForChildViewController:childViewController
+                                   decreaseButton:decreaseButton
+                                   increaseButton:increaseButton];
+}
+- (void)
+    childViewControllerDidRequestPreferredContentSizeCategoryIncrement:
+        (UIViewController *)childViewController
+                                                        decreaseButton:(MDCButton *)decreaseButton
+                                                        increaseButton:(MDCButton *)increaseButton {
+  [self increaseContentSizeForChildViewController:childViewController
+                                   decreaseButton:decreaseButton
+                                   increaseButton:increaseButton];
+}
+
+- (void)childViewControllerDidRequestUserInterfaceStyle:(UIViewController *)childViewController
+                                     userInterfaceStyle:(UIUserInterfaceStyle)userInterfaceStyle
+    API_AVAILABLE(ios(12.0)) {
+  UITraitCollection *userInterfaceStyleTraitCollection =
+      [UITraitCollection traitCollectionWithUserInterfaceStyle:userInterfaceStyle];
+  UITraitCollection *currentTraitCollection = childViewController.traitCollection;
+  NSArray *traitCollections = @[ currentTraitCollection, userInterfaceStyleTraitCollection ];
+  UITraitCollection *traitCollection =
+      [UITraitCollection traitCollectionWithTraitsFromCollections:traitCollections];
+  [self setOverrideTraitCollection:traitCollection forChildViewController:childViewController];
+  [self.view setNeedsLayout];
+}
+
+@end
```

Comment:
Ah, interesting that you chose to move the examples themselves into supplemental as well. Makes me think we're missing a category in our conventions for "things that are functional demos, but not necessarily examples"...

Reply:
Yeah I wasn't sure which ones to move so I moved all of them. And that's an interesting thought...